lib/index.js

```diff
@@ -1,21 +1,65 @@
-export default (callback) => (input) => {
+export default (opts = {}, callback) => (input) => {
   return function watch(log) {
     const chokidar = require('chokidar');
 
+    /* eslint-disable no-param-reassign, immutable/no-mutation */
+
+    if (!callback && typeof opts === 'function') {
+      callback = opts;
+      opts = {};
+    }
+
+    if (typeof opts === 'string' || Array.isArray(opts)) {
+      opts = { files: opts };
+    }
+
+    if (input && input.length && !opts.files) {
+      opts.files = input;
+    }
+
+    opts.events = opts.events || [ 'add', 'change' ];
+
+    /* eslint-enable */
+
     return new Promise((resolve, reject) => {
-      const init = () => {
-        chokidar
-          .watch(input, { persistent: true })
-          .on('change', callback)
-          .on('error', reject)
-          .on('ready', () => {
-            log('press ctrl-c to exit');
-          });
-      };
-
-      callback(input)
-        .then(init)
-        .catch(init);
+
+      log('Globbing files...');
+
+      const initialFileList = [];
+      const initialListener = (file) => initialFileList.push(file);
+
+      const watcher = chokidar.watch(opts.files, { persistent: true, ...opts });
+
+      watcher.once('error', reject);
+
+      watcher.on('add', initialListener);
+      watcher.once('ready', () => {
+        watcher.removeListener('add', initialListener);
```

Comment:
This is similar to how it was before. 

Earlier it was first using `globby` to discover an initial file list. Then it called the `callback` with it. And then finally it starts the chokidar watcher passing it the callback again.

Now it only uses chokidar to do both. Chokidar fires "add" event when it's adding files to its list initially. This is what `initialListener` is used for. Then after it fires "ready" the `initialListener` is removed and a new listener is added.